actions/src/openstack_check_actions.py

```diff
@@ -0,0 +1,277 @@
+                                "os-extended-snapshot-attributes:project_id"
+                            ],
+                        }
+                    )
+                except KeyError:
+                    snap_list.append(
+                        {
+                            "name": snapshot["name"],
+                            "id": snapshot["id"],
+                            "updated": snapshot["created_at"],
+                            "project_id": snapshot["location"]["project"]["id"],
+                        }
+                    )
+        return snap_list
+
+    @staticmethod
+    def create_ticket(
+        tickets_info, email: str, api_key: str, servicedesk_id, requesttype_id
+    ):
+        # pylint: disable=line-too-long
+        """
+        Function to create tickets in an atlassian project. The tickets_info value should be formatted as such:
+        {
+            "title": "This is the {p[title]}",
+            "body": "This is the {p[body]}", #The {p[value]} is used by python formatting to insert the correct value in this spot, based off data passed in with server_list
+            "server_list": [
+                {
+                    "dataTitle":{"title": "This replaces the {p[title]} value!"}, #These dictionary entries are picked up in create_ticket
+                    "dataBody":{"body": "This replaces the {p[body]} value"}
+                }
+            ] This list can be arbitrarily long, it will be iterated and each element will create a ticket based off the title and body keys and modified with the info from dataTitle and dataBody. For an example on how to do this please see deleting_machines_check
+        }
+        """
+        print(tickets_info)
+        try:
+            actual_tickets_info = tickets_info["result"]
+        except TypeError:
+            actual_tickets_info = ast.literal_eval(tickets_info)
+
+        if len(actual_tickets_info["server_list"]) == 0:
+            logging.info("No issues found")
+            sys.exit()
+        for i in actual_tickets_info["server_list"]:
+
+            issue = requests.post(
+                "https://stfc.atlassian.net/rest/servicedeskapi/request",
+                auth=HTTPBasicAuth(email, api_key),
+                headers={
+                    "Accept": "application/json",
+                    "Content-Type": "application/json",
+                },
+                json={
+                    "requestFieldValues": {
+                        "summary": actual_tickets_info["title"].format(
+                            p=i["dataTitle"]
+                        ),
+                        "description": actual_tickets_info["body"].format(
+                            p=i["dataBody"]
+                        ),
+                    },
+                    "serviceDeskId": servicedesk_id,  # Point this at relevant service desk
+                    "requestTypeId": requesttype_id,
+                },
+            )
```

Comment:
Can we move this into a new module in `lib` I can see a `post_ticket` method getting use in other actions too so it would be good to move it before we duplicate it after forgetting we have it